Lib/argparse.py

```diff
@@ -2597,3 +2601,24 @@ def error(self, message):
         self.print_usage(_sys.stderr)
         args = {'prog': self.prog, 'message': message}
         self.exit(2, _('%(prog)s: error: %(message)s\n') % args)
+
+
+class PrettyExecutableName(object):
```

Comment:
Why a class when a function will do? For example, this function:
```python
def adjust_command(cmd):
    """
    Adjust a command line which uses 'python' to substitute the name
    of the actual running interpreter, e.g. 'python3' or 'python3.10'.
    """
    parts = cmd.split(' ', 1)
    exe = parts[0]
    if exe != 'python':
        return cmd
    if exe == 'python':
        sysexe = sys.executable
        if not sysexe:
            return cmd
        exe = os.path.basename(sysexe)
        if os.name == 'nt':
            exe = os.path.splitext(exe)[0]
    return exe if len(parts) == 1 else f'{exe} {parts[1]}'
```
would appear to do the trick and not be restricted to e.g. `-m` commands. You should be able to try it interactively [here](https://replit.com/join/anuixcivrv-vsajip).